lib/ansiblelint/utils.py

```diff
@@ -181,19 +182,24 @@ def find_children(playbook, playbook_dir):
     basedir = os.path.dirname(playbook[0])
     items = _playbook_items(playbook_ds)
     for item in items:
-        for child in play_children(basedir, item, playbook[1], playbook_dir):
-            if "$" in child['path'] or "{{" in child['path']:
-                continue
-            valid_tokens = list()
-            for token in split_args(child['path']):
-                if '=' in token:
-                    break
-                valid_tokens.append(token)
-            path = ' '.join(valid_tokens)
-            results.append({
-                'path': path_dwim(basedir, path),
-                'type': child['type']
-            })
+        try:
+            for child in play_children(basedir, item, playbook[1], playbook_dir):
+                if "$" in child['path'] or "{{" in child['path']:
+                    continue
+                valid_tokens = list()
+                for token in split_args(child['path']):
+                    if '=' in token:
+                        break
+                    valid_tokens.append(token)
+                path = ' '.join(valid_tokens)
+                results.append({
+                    'path': path_dwim(basedir, path),
+                    'type': child['type']
+                })
+        except MatchError as e:
+            e.filename = playbook[0]
+            raise e
```

Comment:
This try/except block is too broad. It should contain only one instruction inside. Since the exception is happening in `play_children()`, it should be either moved there or you could create a decorator-style wrapper that would encapsulate this.